### PR TITLE
Added fixes for _MemIndexed in response to issue #171

### PR DESCRIPTION
**Pull request by nvandervoort**
and some tests for those changes

### CODE REVIEW

File: pyrtl/memory.py [hunk 1]
```diff
@@ -69,6 +70,10 @@ def sign_extended(self, bitwidth):
     def zero_extended(self, bitwidth):
         return as_wires(self).zero_extended(bitwidth)
 
+    @property
+    def name(self):
+        raise PyrtlError("MemIndexed is a temporary object and therefore doesn't have a name")
+
```
Comment (Gamrix) [inline pyrtl/memory.py]: Umm, read the text at the end again. the memindexed should make accesses and changes to a _MemIndexed.name affect the underlying wire, and make the wire if needed.

**Comment by Gamrix** [comment]
For the `test_write_memindexed_ior`, `test_read_memindexed_ior` and `test_read_memindexed_ilshift`, could you create a meaningful simulation (5 sim.steps for each test) to check that the circuit actually performs as intended (this will include getting rid of some of the constants)

File: tests/test_memblock.py [hunk 2]
```diff
@@ -106,49 +106,81 @@ def setUp(self):
 
     def test_memindexed_name(self):
         x = self.mem[2]
-        with self.assertRaises(pyrtl.PyrtlError):
-            print(x.name)
+        x.name = 'test_name'  # setter
+        self.assertEquals(x.name, 'test_name')  # getter
+        # with self.assertRaises(pyrtl.PyrtlError):
+        #     print(x.name)
```
Comment (Gamrix) [inline tests/test_memblock.py]: please check that the wire inside x also gets assigned `'test_name'`

**Comment by Gamrix** [comment]
In order to properly check a simulation, the results of the simulation must be checked through `sim.inspect()`, `sim.inspect_mem()` or `sim_trace.trace`

**Comment by Gamrix** [comment]
Also, can you add an additional unittest to make sure that doing both a memory read as well as setting of a name does not result in multiple read ports being made?

**Comment by nvandervoort** [comment]
For what tests would I need to use `sim.inspect()`, etc.?

**Comment by Gamrix** [comment]
the ones that are being simulated (aka the three I mentioned above).

**Pull request merged**